python/cuspatial/cuspatial/geometry/geocolumn.py

```diff
@@ -0,0 +1,471 @@
+    using native GeoArrowBuffers. These will be used to convert to GeoPandas
+    GeoSeries if necessary.
+    """
+
+    def __init__(self, meta: Union[GeoArrowBuffers, dict]):
+        if isinstance(meta, dict):
+            self.input_types = meta["input_types"]
+            self.input_lengths = meta["input_lengths"]
+            return
+        buffers = meta
+        self.input_types = []
+        self.input_lengths = []
+        if buffers.points is not None:
+            self.input_types.extend(repeat("p", len(buffers.points)))
+            self.input_lengths.extend(repeat(1, len(buffers.points)))
+        if buffers.multipoints is not None:
+            self.input_types.extend(repeat("mp", len(buffers.multipoints)))
+            self.input_lengths.extend(repeat(1, len(buffers.multipoints)))
+        if buffers.lines is not None:
+            if len(buffers.lines.mlines) > 0:
+                self.input_types.extend(repeat("l", buffers.lines.mlines[0]))
+                self.input_lengths.extend(repeat(1, buffers.lines.mlines[0]))
+                for ml_index in range(len(buffers.lines.mlines) // 2):
+                    self.input_types.extend(["ml"])
+                    self.input_lengths += [1]
+                    mline_size = (
+                        buffers.lines.mlines[ml_index * 2 + 1]
+                        - 1
+                        - buffers.lines.mlines[ml_index * 2]
+                    )
+                    self.input_types.extend(repeat("l", mline_size))
+                    self.input_lengths.extend(repeat(1, mline_size))
+            else:
+                self.input_types.extend(repeat("l", len(buffers.lines)))
+                self.input_lengths.extend(repeat(1, len(buffers.lines)))
+        if buffers.polygons is not None:
+            if len(buffers.polygons.mpolys) > 0:
+                self.input_types.extend(
+                    repeat("poly", buffers.polygons.mpolys[0])
+                )
+                self.input_lengths.extend(
+                    repeat(1, buffers.polygons.mpolys[0])
+                )
+                for mp_index in range(len(buffers.polygons.mpolys) // 2):
+                    mpoly_size = (
+                        buffers.polygons.mpolys[mp_index * 2 + 1]
+                        - buffers.polygons.mpolys[mp_index * 2]
+                    )
+                    self.input_types.extend(["mpoly"])
+                    self.input_lengths.extend([mpoly_size])
+                    self.input_types.extend(repeat("poly", mpoly_size))
+                    self.input_lengths.extend(repeat(1, mpoly_size))
+            else:
+                self.input_types.extend(repeat("poly", len(buffers.polygons)))
+                self.input_lengths.extend(repeat(1, len(buffers.polygons)))
+
+    def copy(self, deep=True):
+        return type(self)(
+            {
+                "input_types": self.input_types.copy(),
+                "input_lengths": self.input_lengths.copy(),
```

Comment:
Should the deep parameter be passed to these copies?

Reply:
These are host arrays, so it is not necessary. Nor is the argument necessary. :)